test/jasmine/tests/validate_test.js

```diff
@@ -0,0 +1,109 @@
+        expect(obj.path).toEqual(path);
+
+        // TODO test msg
+    }
+
+    it('should report when trace is defaulted to not be visible', function() {
+        var out = Plotly.validate([{
+            type: 'scatter'
+            // missing 'x' and 'y
+        }], {});
+
+        assertErrorShape(out, [1], 0);
+        assertErrorContent(out.data[0][0], 'invisible', 0, '');
+    });
+
+    it('should report when trace contains keys not part of the schema', function() {
+        var out = Plotly.validate([{
+            x: [1, 2, 3],
+            markerColor: 'blue'
+        }], {});
+
+        assertErrorShape(out, [1], 0);
+        assertErrorContent(out.data[0][0], 'schema', ['markerColor'], '');
+    });
+
+    it('should report when trace contains keys that are not coerced', function() {
+        var out = Plotly.validate([{
+            x: [1, 2, 3],
+            mode: 'lines',
+            marker: { color: 'blue' }
+        }, {
+            x: [1, 2, 3],
+            mode: 'markers',
+            marker: {
+                color: 'blue',
+                cmin: 10
+            }
+        }], {});
+
+        assertErrorShape(out, [1, 1], 0);
+        expect(out.layout.length).toEqual(0);
+        assertErrorContent(out.data[0][0], 'unused', ['marker'], '');
+        assertErrorContent(out.data[1][0], 'unused', ['marker', 'cmin'], '');
+
+    });
+
+    it('should report when trace contains keys set to invalid values', function() {
+        var out = Plotly.validate([{
+            x: [1, 2, 3],
+            mode: 'lines',
+            line: { width: 'a big number' }
+        }, {
+            x: [1, 2, 3],
+            mode: 'markers',
+            marker: { color: 10 }
+        }], {});
+
+        assertErrorShape(out, [1, 1], 0);
+        assertErrorContent(out.data[0][0], 'value', ['line', 'width'], '');
+        assertErrorContent(out.data[1][0], 'value', ['marker', 'color'], '');
+    });
+
+    it('should work with isLinkedToArray attributes', function() {
+        var out = Plotly.validate([], {
+            annotations: [{
+                text: 'some text'
+            }, {
+                arrowSymbol: 'cat'
+            }],
+            xaxis: {
+                type: 'date',
+                rangeselector: {
+                    buttons: [{
+                        label: '1 month',
+                        step: 'all',
+                        count: 10
+                    }, {
+                        title: '1 month'
+                    }]
+                }
+            },
+            shapes: [{
+                opacity: 'none'
+            }]
+        });
+
+        assertErrorShape(out, [], 4);
+        assertErrorContent(out.layout[0], 'schema', ['annotations[1]', 'arrowSymbol'], '');
+        assertErrorContent(out.layout[1], 'unused', ['xaxis', 'rangeselector', 'buttons[0]', 'count'], '');
+        assertErrorContent(out.layout[2], 'schema', ['xaxis', 'rangeselector', 'buttons[1]', 'title'], '');
```

Comment:
> Moreover, I thought also about writing paths to `_isLinkedToArray` attributes as e.g.
> 
> ```
> ['xaxis', 'rangeselector', 'buttons', 1, 'title']
> 
> // instead of
> ['xaxis', 'rangeselector', 'buttons[1]', 'title']
> ```

I added your missing comma, but yes, if you're going to return an array like that, integers for array indices is the way to go so you could just loop through with `container = container[path[i]]`. I would think though that, aside from our own internal uses (which can certainly use `Lib.nestedProperty` unless we export these results to another language), folks will rarely want programmatic access to these items. Mostly they'd just want something easy to look at and spot their error, in which case I'd think the single string version is nicer.